src/datachain/fs/utils.py

```diff
@@ -0,0 +1,30 @@
+from typing import TYPE_CHECKING
+
+from fsspec.implementations.local import LocalFileSystem
+
+if TYPE_CHECKING:
+    from fsspec import AbstractFileSystem
+
+
+def _isdir(fs: "AbstractFileSystem", path: str) -> bool:
+    info = fs.info(path)
+    return info["type"] == "directory" or (
+        info["size"] == 0 and info["type"] == "file" and info["name"].endswith("/")
+    )
```

Comment:
I extract this out so that if we need to implement `isdir`, it would not be too difficult. 

We cannot implement `isdir` to be a negative of `isfile` as it misses non-existing file.